node/libs/storage/src/lib.rs

```diff
@@ -35,46 +57,50 @@ pub struct Storage {
     block_writes_sender: watch::Sender<BlockNumber>,
 }
 
-impl Storage {
+impl RocksdbStorage {
     /// Create a new Storage. It first tries to open an existing database, and if that fails it just creates a
     /// a new one. We need the genesis block of the chain as input.
     // TODO(bruno): we want to eventually start pruning old blocks, so having the genesis
     //   block might be unnecessary.
-    pub fn new(genesis_block: &validator::FinalBlock, path: &Path) -> Self {
+    pub async fn new(
+        ctx: &ctx::Ctx,
+        genesis_block: &validator::FinalBlock,
+        path: &Path,
+    ) -> anyhow::Result<Self> {
+        scope::run!(ctx, |_, scope| async {
+            Ok(scope
+                .spawn_blocking(|| Self::new_blocking(genesis_block, path))
+                .join(ctx)
+                .await?)
+        })
+        .await
+    }
+
+    /// Blocking version of [`Self::new()`].
+    fn new_blocking(genesis_block: &validator::FinalBlock, path: &Path) -> anyhow::Result<Self> {
```

Comment:
why do we need the blocking version of new? Is it so that a single blocking thread is used for all initial db accesses?